editor/src/clj/editor/gui.clj

```diff
@@ -827,18 +829,29 @@
 
 (def ^:private validate-texture (partial validate-optional-gui-resource "texture '%s' does not exist in the scene" :texture))
 
+(g/defnk produce-shape-base-node-msg [visual-base-node-msg manual-size size-mode texture clipping-mode clipping-visible clipping-inverted]
+  (assoc visual-base-node-msg
+    :size (v3->v4 manual-size)
+    :size-mode size-mode
+    :texture texture
+    :clipping-mode clipping-mode
+    :clipping-visible clipping-visible
+    :clipping-inverted clipping-inverted))
+
 (g/defnode ShapeNode
   (inherits VisualNode)
 
-  (property size types/Vec3 (default [0 0 0])
-            (value (g/fnk [size size-mode texture-size]
+  (property manual-size types/Vec3 (default [200.0 100.0 0.0])
+            (dynamic visible (g/constantly false)))
+  (property size types/Vec3
+            (value (g/fnk [manual-size size-mode texture-size]
                           (if (= :size-mode-auto size-mode)
-                            (or texture-size size)
-                            size)))
-            (dynamic read-only? (g/fnk [size-mode type] (and (or (= type :type-box) (= type :type-pie))
-                                                             (= :size-mode-auto size-mode)))))
+                            (or texture-size manual-size)
+                            manual-size)))
+            (set (fn [_evaluation-context self _old-value new-value]
+                   (g/set-property self :manual-size new-value)))
+            (dynamic read-only? (g/fnk [size-mode] (= :size-mode-auto size-mode))))
```

Comment:
The `size` property is what we display to users. It is backed by the `manual-size` property, but will show the size of the selected image when you use `:size-mode-auto`. Note that the `node-msg` output only depends on the `manual-size` property and not `size`. This eliminates the data dependency between the image size and the `save-data` of our gui nodes.